Mathlib/GroupTheory/SpecificGroups/ZGroup.lean

```diff
@@ -94,4 +99,46 @@ instance isCyclic_abelianization [Finite G] [IsZGroup G] : IsCyclic (Abelianizat
 
 end Nilpotent
 
+section Classification
+
+/-- An extension of coprime Z-groups is a Z-group. -/
+theorem isZGroup_of_coprime
+    [Finite G] [IsZGroup G] [IsZGroup G''] (h_le : f'.ker ≤ f.range)
+    (h_cop : (Nat.card G).Coprime (Nat.card G'')) : IsZGroup G' := by
```

Comment:
This proof is slightly on the long side, but I didn't see a clear opportunity to split.

Reply:
I managed to find a nice split, I think.